### DISCUSSION TITLE
update e-kasa

### DISCUSSION THREAD

**Discussion by PeterI123** [datapacsro/oec#34] [Q&A] [answered]
hello / ahoj
I have a device with old firmware / mám zariadenie so starým FW:
OEC.LAN Standalone v1.0.7.0 update 010
"SerialNumber": "20010379",
"swid": "77CD92682012DE2205502BE0FCBE0019B420D7F5

How can I update to the latest firmware as quickly as possible / ako môžem čo najrýchlešie updatovať na najnovší FW:
OEC.LAN Standalone v1.5.0.0 update 017
60A1929729828DAA625B70A066909B9078F8FB81

Since there are about 24 updates, is it possible to update it with some cumulative updates? / Je tam cca 24 updatov. je možné to updatovať nejakými kumulatívnymi updatami?

link:
https://github.com/datapacsro/oec/wiki/Ako-zisti%C5%A5-verziu-software-OEC

thank you
Peter I.
Bratislava / Slovensko

**Comment by slado** [comment, marked as answer]
Zdravim Peter,

ak mate update 010, tak podla releases sa mozete dostat na aktualnu verziu takto
10->16->17
Cize staci na 2 kroky. Aktualizacie najdete tu https://github.com/datapacsro/oec/releases 

S pozdravom

Roman Sladek

**Reply by PeterI123** [reply]
Dík za info, dám vedieť, ako to dopadlo. 
Peter

**Reply by PeterI123** [reply]
dopadlo to dobre :) a teraz som si všimol že už je aj OEC LAN Standalone update 018, no neviem, kde mám nájsť príslušný ZIP subor a súbor s kódom (link ktorý som našiel, obsahuje len ZIP súbor a má názov oec-oeclans018, čo sa mi nezdá, lebo 17-ka mala názov update_oec_lan_standalone_0017). A dúfam, že keď už mám update na 17-ku, ze mozem nahodiť už aj 18-ku. Vopred ďakujem za odpoveď. 
Peter

**Reply by PeterI123** [reply]
a ešte mám otázočku:
ak som správne pochopil tak v konfigurácii e-kasy sa dá nastaviť mail, ktorý prípadne pošle nejaké info o stave kasy a mna zaujímalo, aké info to posiela?
- ak sa urobí update? 
- ak sa zmení konfigurácia? 
- ak sa nahrajú nové certifikáty? 
- informuje e-kasy týmto mailom, že končí platnosť certifikátov? a koľko v predstihu? 
- resp. kde nájdem nejaké info, čo všetko takto nastavený mail na e-kase posiela? 

V manuály čo mám, som o tomto nič nenašiel.
moj fir. mail: peter.istok@bvsas.sk

**Comment by slado** [comment]
Dobry den, 

verzia 018 este nie je schvalena Financnou spravou. Prave prebieha overovanie na ich strane.
Budeme posielat informaciu ked bude verzia 018 dostupna.
OEC neposiela e-mail o svojom stave, ale umoznuje aby ste monitorovali stav OEC pomocou mechanizmu healthcheck. na http://<adresa>/oec/health viete ziskat aktualny stav a podla toho reagovat. Mozete vyuzit existujuce open-source monitorovacie nastroje ako Zabbix alebo Prometheus na monitoring a alerting. Viac o healthcheck tu [OEC Healthcheck.docx](https://github.com/datapacsro/oec/raw/oeclans018/OEC%20Healthcheck.docx)

S pozdravom

Roman Sladek

**Reply by PeterI123** [reply]
Ďakujem za info ohladne ver. 18-ky. 
Čo sa týka toho mailu, čo sa dá nastaviť v konf. pokladní/editácia pokladne, tak nerozumiem, prečo sa tam dá  vobec mail definovať, ked nič neposiela? 
Čo by ste mi vedel poradiť, ako by som ziskal informáciu, že mi bude končiť platnosť certifikátov na e-kase?

**Comment by slado** [comment]
Konfiguracia e-mail servera sluzi na odosielanie dokladov mailom namiesto tlace.
pri registracii dokladu mozete specifikovat ci sa ma tlacit alebo odoslat mailom v ReceiptRenderer https://datapacsro.github.io/oec/#api-Default-addReceipt

Ak chcete sledovat platnost certifikatov, odporucam Zabbix, Prometheus alebo iny tool, ktory bude v pravidelnych intervaloch (raz denne napriklad) zistovat stav OEC a na zaklade toho posielat alerty.

**Reply by PeterI123** [reply]
ďakujem za info. a vedel by ste mi upresniť/alebo pomocť s nejakym navodom, ako sa dá sledovať blížiacia sa  platnosť certifikátov cez Zabbix? ďakujem

**Reply by speedo919** [reply]
@PeterI123  Vaše otázky sú základné otázky ohladne ekasy Datapac. Ako prvé Vám doporučujem naštudovať si dokumentáciu ktorá je tu na githube a zodpovedá všetky vaše otázky.

**Reply by slado** [reply]
Dobry den,

Radi Vam poskytneme konzultacie ohladne Zabbix alebo Prometheus na komercnej baze. Ak mate zaujem, dajte mi vediet.
Dakujem

S pozdravom

Roman Sladek

**Comment by PeterI123** [comment]
Pekný deň prajem, 
e-kasu som mal updatovanu v. 017 a teraz som ju updatoval na v. 018
Všetko prebehlo OK a vrátane sucess":true. No "swid" je : "7F08506D1B9FF0481F68D8FDF6552823DAED7E1E" a vy máte v popise, že má byť: 60A1929729828DAA625B70A066909B9078F8FB81, čo sa mi nezdá, lebo rovnaké swin máte uvedené aj u v 017

Mám to teda správne a Vy ste omylom neprepísali swid pre v 018 ?

Dik Peter

**Reply by slado** [reply]
Dakujem za upozornenie. Bolo to chybne vo wiki. Uz je to fixnute.

**Reply by PeterI123** [reply]
dik za opravu :) a ešte teda otazka:
kedže som updatoval zo 17 na 18, tak sa musím venovať aj týmto dvom updatom? 

OEC.LAN Internal v1.5.0.1 update 015
a
OEC.USB CHDU.sk v1.5.0.1 update 015

Dik za odpoveď 
Peter

**Comment by slado** [comment]
Nie, vy pouzivate OEC.LAN Standalone verziu, cize updatom pre ine verzia sa nemusite venovat.

**Reply by PeterI123** [reply]
dik za info
